web/app/components/thread/thread.jsx

```diff
@@ -76,26 +50,21 @@ export default class Thread extends Component {
           onCollapseToggle={this.onCollapseToggle}
         />
 
-        {!collapsed &&
-          !!replies.length &&
-          replies.map(thread => (
-            <Thread
-              key={thread.comment.id}
-              data={thread}
-              mods={{ level: mods.level < 5 ? mods.level + 1 : mods.level }}
-              onReply={props.onReply}
-              onEdit={props.onEdit}
-            />
-          ))}
+        {threadComments}
       </div>
     );
   }
 }
 
-function getCollapsedComments() {
-  return JSON.parse(localStorage.getItem(LS_COLLAPSE_KEY) || '[]');
-}
+const mapStateToProps = (state, props) => ({
+  collapsed: getThreadIsCollapsed(state, props.data.comment),
+});
 
-function saveCollapsedComments(comments) {
-  localStorage.setItem(LS_COLLAPSE_KEY, JSON.stringify(comments));
-}
+const mapDispatchToProps = { setCollapse };
+
+const enhance = connect(
+  mapStateToProps,
+  mapDispatchToProps
+);
+
+export default enhance(Thread);
```

Comment:
I also prefer inline declarations here as well:
```js
export default connect(
  (state, props) => ({
    collapsed: getThreadIsCollapsed(state, props.data.comment)
  }),
  { setCollapse }
)(Thread);
```

Reply:
It will become unreadable as soon as we wrap it up with one more HOC :)

Reply:
✅